Lib/test/test_exceptions.py

```diff
@@ -1285,6 +1285,18 @@ def g():
                 next(i)
                 next(i)
 
+    def test_assert_shadowing(self):
+        # Shadowing AssertionError would cause the assert statement to
+        # misbehave.
+        globals()['AssertionError'] = TypeError
+        global AssertionError
+        try:
+            assert False, 'hello'
+        except Exception as e:
+            del AssertionError
```

Comment:
You need to remove it also if no exception is raised, or non-Exception exception is raised.

Reply:
You are right.